tests/test-coauthors-guest-authors.php

```diff
@@ -124,22 +124,15 @@ public function test_get_guest_author_thumbnail() {
 
 		$this->assertNull( $guest_author_obj->get_guest_author_thumbnail( $guest_author, 0 ) );
 
-		// Checks when guest author has thumbnail.
-		$filename = rand_str() . '.jpg';
-		$contents = rand_str();
-		$upload   = wp_upload_bits( $filename, null, $contents );
-
-		$this->assertTrue( empty( $upload['error'] ) );
-
-		$attachment_id = $this->_make_attachment( $upload );
+		$attachment_id = $this->factory->attachment->create_upload_object( __DIR__ . '/dummy-attachment.png' );
 
 		set_post_thumbnail( $guest_author->ID, $attachment_id );
 
 		$thumbnail = $guest_author_obj->get_guest_author_thumbnail( $guest_author, 0 );
 
 		$this->assertContains( 'avatar-0', $thumbnail );
-		$this->assertContains( $filename, $thumbnail );
-		$this->assertContains( 'src="' . wp_get_attachment_url( $attachment_id ) . '"', $thumbnail );
+		$this->assertContains( 'dummy-attachment', $thumbnail );
```

Comment:
I wondered why this should not be `dummy-attachment.png` but after running a few tests and `var_dump`'ing `$thumnail`, I can understand your intention. If we keep testing, `$thumnail` will be added with a suffix number like `dummy-attachment-7.png`. 

Nitpick: It would be nice to have a comment here.

Reply:
This could also be changed to use `assertStringContainsString()`.

Reply:
> If possible, it would be helpful if there was an explanation why this PR changes the way of creating attachments.

Thanks for your feedback, @htdat. Regarding adding an explanation about this change, I wonder what the benefit of such an explanation would be. It's my understanding that comments should be added to parts that do not explain themselves. In this case, once the PR got merged, a developer will only see that a dummy image will be imported, but will not see that an artificial file had been created in an earlier version. I might be wrong in this case, but I'd not necessarily add an explanation here as the code looks self explaining to me.

> I wondered why this should not be dummy-attachment.png but after running a few tests and var_dump'ing $thumnail, I can understand your intention. If we keep testing, $thumnail will be added with a suffix number like dummy-attachment-7.png.

> Nitpick: It would be nice to have a comment here.

`dummy-attachment-n.png` is indeed the reason why I'm only checking for the filename in this case.

> This could also be changed to use `assertStringContainsString()`.

After your comment, I found the following info regarding `assertContains()` on https://github.com/sebastianbergmann/phpunit/issues/3422#issuecomment-442585223:

> [Soft Deprecation](https://github.com/sebastianbergmann/phpunit/wiki/Release-Process#stage-1-soft-deprecation) in PHPUnit 7.5
[Hard Deprecation](https://github.com/sebastianbergmann/phpunit/wiki/Release-Process#stage-2-deprecation) in PHPUnit 8
[Removal](https://github.com/sebastianbergmann/phpunit/wiki/Release-Process#stage-3-removal) in PHPUnit 9

Therefore, I went ahead and replaced all occurrences of `assertContains()` with `assertStringContainsString()`.

Reply:
> Thanks for your feedback, @htdat. Regarding adding an explanation about this change, I wonder what the benefit of such an explanation would be. It's my understanding that comments should be added to parts that do not explain themselves.

@nielslange I mean an explanation in this PR description for changing the way of creating attachments. Agree that you do not need to add code comments for this. 

> dummy-attachment-n.png is indeed the reason why I'm only checking for the filename in this case.

The code comment is what I meant for this part.

Reply:
Ah, that's what you meant. Adding an explanation about the change to the PR makes sense. I'll add one now.